processing/src/main/java/io/druid/granularity/AllGranularity.java

```diff
@@ -62,7 +62,7 @@ public boolean equals(Object o)
   @Override
   public int hashCode()
   {
-    return 0;
+    return getClass().hashCode();
```

Comment:
ok, then we can just make it 0 for none and 1 for all, if collisions are a problem. Seems unnecessary to use class name.